src/index.js

```diff
@@ -45,3 +45,22 @@ exports.Queue = Plotly.Queue;
 
 // export d3 used in the bundle
 exports.d3 = require('d3');
+
+Plotly.register({
+    traces: {
+        bar: require('./traces/bar'),
```

Comment:
> I don't like using arguments unless it's the only way because it isn't as easily portable (you would have to use Array.spread)

Wouldn't [`arguments.length`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments/length) by enough? 

@mdtusz that said, I'm ok with `Plotly.register` taken a single array as arguments. Your call.

> Can we go with registerTraces as the function name,

I'd vote for a single `Plotly.register` and taking care of the distinction between trace and component modules behind the scene.